core/runtime/src/main/java/io/quarkus/runtime/Quarkus.java

```diff
@@ -248,7 +232,13 @@ public static void manualInitialize() {
             Class<?> appClass = Class.forName("io.quarkus.runner.ApplicationImpl");
             manualApp = (Application) appClass.getDeclaredConstructor().newInstance();
             manualState = MANUAL_INITIALIZED;
-            Core.getGlobalContext().register(new QuarkusCracBootstrapResource());
+            if (CracRecorder.enabled) {
+                // NOTE: I tried to do this within a recorder method, but the resource was never registered
+                // with Crac.  Ideally this would be registered within CracRecorder.register()
```

Comment:
Hmm this is very strange; `manualInitialize` instantiates the application, which should cause `ApplicationImpl` to be initialized which should in turn run all `STATIC_INIT` generated code.

Reply:
Yeah, putting in System.out.printlns() shows that static initialization code was being called, but, the CRAC callbacks didn't happen.